### PR TITLE
ci: add buf push github action

### PR DESCRIPTION
**Pull request by bojand**
Adds a manual dispatch GitHub Action to push the protos to Bug Schema Registry.

### CODE REVIEW

File: .github/workflows/bsr-push.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      - uses: bufbuild/buf-breaking-action@v1
+        with:
+          against: https://github.com/acme/weather.git#branch=main,ref=HEAD~1,subdir=proto
+      # Push module to the BSR
+      - uses: bufbuild/buf-push-action@v1
+        with:
+          input: proto
+          buf_token: ${{ secrets.BUF_TOKEN }}
+          create_visibility: private
+          draft: ${{ github.ref_name != 'main'}}
```
Comment (weeco) [inline .github/workflows/bsr-push.yml:30]:
```suggestion
          draft: ${{ github.ref_name != 'master'}}
```

File: .github/workflows/bsr-push.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+          input: proto
+      # Run breaking change detection for Protobuf sources against the current `main` branch
+      - uses: bufbuild/buf-breaking-action@v1
+        with:
+          against: https://github.com/acme/weather.git#branch=main,ref=HEAD~1,subdir=proto
+      # Push module to the BSR
+      - uses: bufbuild/buf-push-action@v1
+        with:
+          input: proto
+          buf_token: ${{ secrets.BUF_TOKEN }}
```
Comment (weeco) [inline .github/workflows/bsr-push.yml:28]: Did you create the secret in GH action?

File: .github/workflows/bsr-push.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        with:
+          github_token: ${{ github.token }}
+      # Lint Protobuf sources
+      - uses: bufbuild/buf-lint-action@v1
+        with:
+          input: proto
+      # Run breaking change detection for Protobuf sources against the current `main` branch
+      - uses: bufbuild/buf-breaking-action@v1
+        with:
+          against: https://github.com/acme/weather.git#branch=main,ref=HEAD~1,subdir=proto
```
Comment (weeco) [inline .github/workflows/bsr-push.yml:23]: Wrong repo?

File: .github/workflows/bsr-push.yml (hunk 2 above)
Comment (bojand) [reply .github/workflows/bsr-push.yml:36]: No, I do not seem to have permissions to add tokens to actions.

Technically we should have a non-personal / security managed BSR user that the token belongs to. But I guess to start with a personal token can do.

Comment (bojand) [reply .github/workflows/bsr-push.yml:36]: I've been told the secret has been added.

File: .github/workflows/bsr-push.yml [hunk 4]
```diff
@@ -0,0 +1,38 @@
+name: Push to Buf Schema Registry
+
+on:
+  push:
+    branches:
+      - master
+    paths:
+      - proto/**.proto
+  workflow_dispatch:
```
[the remark this answers is not in the document]
Comment (bojand) [reply .github/workflows/bsr-push.yml:9]: This is so we can run the action manually if we need to

**Review by weeco — APPROVED** [review]

**Pull request merged**